src/main/java/org/terasology/las/yingyang/YingYangRasterizer.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.las.yingyang;
+
+import org.terasology.ligthandshadow.componentsystem.LASUtils;
+import org.terasology.math.ChunkMath;
+import org.terasology.math.Region3i;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.registry.CoreRegistry;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.chunks.CoreChunk;
+import org.terasology.world.generation.Region;
+import org.terasology.world.generation.WorldRasterizerPlugin;
+import org.terasology.world.generator.plugin.RegisterPlugin;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.BooleanSupplier;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static java.util.Collections.singletonMap;
+
+@RegisterPlugin
+public class YingYangRasterizer implements WorldRasterizerPlugin {
```

Comment:
Is it possible to use structural templates in the world gen phase? For static structures like this it might be a better fit that generation in code...

Reply:
No, not possible, I discussed with jellysnake. He said it is not yet supported.